src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/squeeze.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v0 {
+
+/**
+ * \brief Do Squeeze shape inference.
+ *
+ * \tparam T             Type of input/output shapes.
+ *
+ * \param op             Squeeze operator pointer.
+ * \param input_shapes   Squeeze input shapes.
+ * \param output_shapes  Output shapes result of squeeze shape inference.
+ * \param constant_data  Map of constant data.
+ */
+template <class T>
+void shape_infer(const Squeeze* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
+
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 1 || input_shapes.size() == 2) && output_shapes.size() == 1);
+
+    const auto& arg_shape = input_shapes[0];
+    auto& output_shape = output_shapes[0];
+
+    bool has_static_axes;  // empty axis is also treat as static
+    std::set<int64_t> unique_axes;
+    if (arg_shape.rank().is_static() && (input_shapes.size() == 2)) {
+        const auto& axes_shape = input_shapes[1];
+
+        NODE_VALIDATION_CHECK(op,
+                              axes_shape.is_dynamic() || is_rank_compatible_any_of(axes_shape.rank(), {0, 1}),
+                              "Second input (axes) should not be of rank higher than 1. Got: ",
+                              axes_shape.rank().get_length());
+
+        std::vector<int64_t> axes;
+        has_static_axes = axes_shape.is_static() && get_data_as_int64<T>(1, op, axes, constant_data);
+        if (has_static_axes) {
+            normalize_axes(op, arg_shape.rank().get_length(), axes);
+            unique_axes = std::set<int64_t>(axes.cbegin(), axes.cend());
+        }
+    } else {
+        has_static_axes = true;
+    }
+
+    if (arg_shape.rank().is_static() && has_static_axes) {
+        std::vector<DimType> out_dims;
+        out_dims.reserve(arg_shape.rank().get_length());
```

Comment:
No, it is to reserve memory only (to avoid allocations during copy) to fill data instead set and overwrite. The final size will set after conditional copy of data.